### ISSUE TITLE
QuestionnaireStepper not scrollable

### ISSUE THREAD

**Issue by StefanKlose** [sujrd/faiadashu#146]
Hi,

I'm using Faiadashu (ref: v0.11.9) and the QuestionnaireScroller is working as expected. I have some issues with the QuestionnaireStepper (or I'm using it wrong). Either way it would be nice if you could provide some hints or a solution.

The actual QuestionnaireStepper page does not scroll (vertically). This causes renderflex errors if there are a lot of answer options and/or you deploy for smaller devices. And of course some of the content is not visible.

I tried to solve this by wrapping a SingleChildScrollview around the Stepper child in the DefaultQuestionnaireScaffoldBuilder.  Here is my code:

```
Expanded _buildScrollableQuestionnairePage(double height, double width, Widget child) {
    return Expanded(
        child: SingleChildScrollView(
            controller: this.scrollController, child: SizedBox(height: height, width: width, child: child)));
  }
```

As you can see I have to provide a height for the SizedBox, otherwise it does not work at all (error without SizedBox: RenderFlex children have non-zero flex but incoming height constraints are unbounded).

But I don't know the size of the child and every possible solution I tried (from StackOverflow/Google) results in errors or only returns the size of the entire canvas and not the actual child.

Is there a fix in a future Faiadashu revision that implements vertical scrolling for QuestionnaireSteppers?

Or do you have any idea how to get the actual size of the child that is rendered by the DefaultQuestionnaireScaffoldBuilder?

kind regards
Dr. Stefan Klose

**Comment by fjwong** [comment]
@StefanKlose Thank you for the report. 

The QuestionnaireStepper widget itself does not implement scrolling by default on each page. This is done on purpose so that we can have more freedom when customizing the page layout via a QuestionnaireTheme.

That said, it appears the default QuestionnaireTheme was not made with vertical scrolling in mind for stepper pages. This is being addressed in #147. If you have some time, please try it and let us know if it works for you.

**Comment by StefanKlose** [comment]
thank you very much. It is working now as intended :-)

I did not use another branch, I simply overrode stepperPageItemBuilder of QuestionnaireThemeData with the following code (also thanks @bennyhaeusler)

  ```
static Widget _defaultStepperPageItemBuilder(
    BuildContext context,
    QuestionnaireItemFiller itemFiller,
  ) {
    return Column(
      children: [
        Flexible(
          child: SingleChildScrollView(
            child: Container(
              padding: const EdgeInsets.symmetric(horizontal: 16.0),
              child: itemFiller,
            ),
          ),
        ),
      ],
    );
  }
```

kind regards
Dr. Stefan Klose

**Comment by fjwong** [comment]
Fixed in #147
